internal/db/repo/scheduler.go

```diff
@@ -0,0 +1,201 @@
+package repo
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/dimkene/go_final_project/internal/dateutil"
+	"github.com/dimkene/go_final_project/internal/models"
+)
+
+// чтобы оперировать Tasks (TaskCreationRequest), нужна всегда ссылка на БД
+type TasksRepository struct {
+	db *sql.DB
+}
+
+func NewTasksRepository(db *sql.DB) TasksRepository {
+	return TasksRepository{db: db}
+}
+
+func (tr TasksRepository) AddTask(t models.Task) (int, error) {
+	task, err := tr.db.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES (:date, :title, :comment, :repeat)",
+		sql.Named("date", t.Date),
+		sql.Named("title", t.Title),
+		sql.Named("comment", t.Comment),
+		sql.Named("repeat", t.Repeat))
+
+	if err != nil {
+		return 0, err
+	}
+
+	id, err := task.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+
+	// возвращаем идентификатор последней добавленной записи
+	return int(id), nil
+}
+
+func (tr TasksRepository) PostTaskDone(id int) (*models.Task, error) {

```

Comment:
БД ничего не знает про тип запросов и как там все наверху идет
```suggestion
func (tr TasksRepository) TaskDone(id int) (*models.Task, error) {

```